### ISSUE TITLE
Support for <WarnOn> MSBuild property

### ISSUE THREAD

**Issue by NinoFloris** [JetBrains/resharper-fsharp#379]
F# 6.0 SDK released with support for specifying warnings via a new MSBuild property which isn't picked up by Rider yet.

I'm assuming to light this up, support needs to be added over here? https://github.com/JetBrains/resharper-fsharp/blob/b3bdd454313fb4b32e924ad5b72238245f020f7c/ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Stages/FSharpCompilerWarningProcessor.fs

**Comment by auduchinok** [comment]
@NinoFloris Could you please share a small repro solution showing where Rider doesn't show a warning while the console build does?

The example from https://github.com/dotnet/fsharp/issues/13437 doesn't produce any additional warning during build for me.

I've also checked it with `FS3180` and got no warnings produced by FCS inside neither Rider nor Visual Studio, so it'd be better to report it to FCS.

**Comment by auduchinok** [comment]
The `FS3180` case is now reported to FCS: https://github.com/dotnet/fsharp/issues/13442.

**Comment by NinoFloris** [comment]
I see, I've changed the repro in the F# issue, there is apparently more broken in the compiler around 3395.

You should now see a warning appear under `StringValues("1")` when 3395 is specified via OtherFlags, but none via WarnOn

**Comment by NinoFloris** [comment]
@auduchinok can you confirm the example repros?

**Comment by auduchinok** [comment]
@NinoFloris Yes, I can repro the `OtherFlags` vs `WarnOn` difference, thank you!
